libpromises/generic_agent.c

```diff
@@ -989,6 +990,34 @@ void CloseLog(void)
 {
     closelog();
 }
+
+static bool SetGroupToSystemGroup()
```

Comment:
Would be great to add a doxygen-like comment before the function describing the behavior (move the comment from the function) and the return value.

Reply:
@vpodzime look good?